pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -121,8 +117,29 @@ public ConsumerBuilderImpl(PulsarClientImpl client, Schema<T> schema) {
         }
         if(conf.isRetryEnable() && conf.getTopicNames().size() > 0 ) {
             TopicName topicFirst = TopicName.get(conf.getTopicNames().iterator().next());
-            String retryLetterTopic = topicFirst.getNamespace() + "/" + conf.getSubscriptionName() + RetryMessageUtil.RETRY_GROUP_TOPIC_SUFFIX;
-            String deadLetterTopic = topicFirst.getNamespace() + "/" + conf.getSubscriptionName() + RetryMessageUtil.DLQ_GROUP_TOPIC_SUFFIX;
+            String retryLetterTopic = topicFirst + "-" + conf.getSubscriptionName() + RetryMessageUtil.RETRY_GROUP_TOPIC_SUFFIX;
+            String deadLetterTopic = topicFirst + "-" + conf.getSubscriptionName() + RetryMessageUtil.DLQ_GROUP_TOPIC_SUFFIX;
+
+            //Issue 9327: do compatibility check in case of the default retry and dead letter topic name changed
+            String oldRetryLetterTopic = topicFirst.getNamespace() + "/" + conf.getSubscriptionName() + RetryMessageUtil.RETRY_GROUP_TOPIC_SUFFIX;
+            String oldDeadLetterTopic = topicFirst.getNamespace() + "/" + conf.getSubscriptionName() + RetryMessageUtil.DLQ_GROUP_TOPIC_SUFFIX;
+            try {
+                PartitionedTopicMetadata partitionedTopicMetadata = client.getLookup().getPartitionedTopicMetadata(TopicName.get(oldRetryLetterTopic)).get();
+                if (partitionedTopicMetadata.partitions > 0) {
+                    retryLetterTopic = oldRetryLetterTopic;
+                }
+            } catch (Exception e) {
+                // oldRetryLetterTopic not exist, do nothing
```

Comment:
can we check for the exact error ?
otherwise even a network error may fall into this case

Reply:
change lookup method so that it could have a retry in case of  retriable error